src/opt/Optimizer/Optimizer.hs

```diff
@@ -128,3 +129,30 @@ dropBorrowBlocks = extend dropBorrowBlock
            ,decls = [([VarNoType name], target)]
            ,body}
       dropBorrowBlock e = e
+
+forwardGeneral = extend forwardGeneral'
+  where
+    forwardGeneral' e@(Forward{forwardExpr=MessageSend{}}) = e
```

Comment:
@kikofernandez's comment is wrong. These two cases explicitly ignore changes to two `forward` cases. The third case does the work for all other `forward` cases. The final case is a catch all for all other kinds of expressions.